js/piwik.js

```diff
@@ -3749,11 +3751,14 @@ if (typeof window.Matomo !== 'object') {
                     (charSet ? '&cs=' + encodeWrapper(charSet) : '') +
                     '&send_image=0';
 
-                var browserFeatures = detectBrowserFeatures();
-                // browser features
-                for (i in browserFeatures) {
-                    if (Object.prototype.hasOwnProperty.call(browserFeatures, i)) {
-                        request += '&' + i + '=' + browserFeatures[i];
+                //Browser FingerPrint
+                if(fingerPrintEnable) {
+                    var browserFeatures = detectBrowserFeatures();
```

Comment:
there are also other places that use `detectBrowserFeatures`. It's probably best to simply return an empty object maybe or so in `detectBrowserFeatures` if the feature is not supposed to be used. We basically don't want to read any of these features when it's disabled.